app/controllers/users_controller.rb

```diff
@@ -355,6 +355,8 @@ def check_signup_allowed(email = nil)
                    domain_mx_servers(domain)
                  end
 
+    return true if Acl.allow_account_creation(request.remote_ip, :domain => domain, :mx => mx_servers)
```

Comment:
I think blocking should have higher priorty (to be evaluated first) than allowing

Reply:
If the IP address is being blocked due to rate limiting, there would be no point allowing it afterwards.

Reply:
Re-arranged to account for non-rate-based blocks first.